examples/05_obstacle_type_qvi/solver_comparison/thermoforming_fixed_point.jl

```diff
@@ -0,0 +1,145 @@
+# g and its derivative
+q = 0.01
+function g(s)
+    if s ≤ 0
+        return 1.0
+    elseif 0 < s < q
+        return (1-s/q)
+    else
+        return 0.0
+    end
+end
+function dg(s)
+    if s ≤ 0
+        return 0.0
+    elseif 0 < s < q
+        return -1/q
+    else
+        return 0.0
+    end
+end
+
+plus(x) = max(0, x)
+plus²(x) = plus(x)^2
+function dplus(x)
+    if x>0
+        return 1.0
+    else
+        return 0.0
+    end
+end
+
+au(u, v) = ∫(∇(u) ⋅ ∇(v)  - fh ⋅ v + γ*(plus ∘ (u - (Φ₀ + φ ⋅ Th))) ⋅ v) * dΩ
+jacu(u, du, v) =∫( ∇(du) ⋅ ∇(v) + γ*(dplus ∘ (u - (Φ₀ + φ ⋅ Th))) ⋅ du ⋅ v) * dΩ
+aT(T, R) =∫( ∇(T) ⋅ ∇(R) + T ⋅ R  - (g ∘ (Φ₀ + φ ⋅ T - uh)) ⋅ R) * dΩ
+jacT(T, dT, R) =∫( ∇(dT) ⋅ ∇(R) + dT ⋅ R - ((dg ∘ (Φ₀ + φ ⋅ T - uh)) ⋅ (φ ⋅ dT)) ⋅ R) * dΩ
+
+au0(u, v) = ∫(∇(u) ⋅ ∇(v) + u ⋅ v) * dΩ
+jacu0(u, du, v) =∫(∇(du) ⋅ ∇(v) + du ⋅ v) * dΩ
+
+opu = FEOperator(au, jacu, Uu, Vu)
+opT = FEOperator(aT, jacT, UT, VT)
+
+energy(u) = ∫(0.5*∇(u) ⋅ ∇(u) - fh ⋅ u) * dΩ
+penalty(u) = ∫(γ/2*(plus² ∘ (u - (Φ₀ + φ ⋅ Th)))) * dΩ
+function update_γ(γ, k)
+    infeasibility = sum(penalty(uh))
+    functional = sum(energy(uh))
+    print("         infeasibility: $infeasibility, functional: $functional.\n")
+    Eₖ = γ * infeasibility / functional
+    θₖ = functional + infeasibility
+    C₂ₖ = Eₖ * (Eₖ + γ) * θₖ / γ
+    C₁ₖ = C₂ₖ / Eₖ
+    τₖ = 1/k
+    return C₂ₖ /(τₖ  * abs(C₁ₖ - θₖ)) - Eₖ
+end
+
+function path_following_solve(opu, solver_u, u, J, γ)
+    newton_its_u = 0
+    u_ = u.free_values[:]
+    for k in 1:100
+        print("         Considering γ=$γ.\n")
+        u, its = solve!(u,solver_u,opu)
+        newton_its_u += its.result.iterations
+        d = uh.free_values[:] - u_
+        cauchy = sqrt(d' * J * d)
+        print("         Cauchy norm: $(cauchy[end]).\n")
+        u_ = u.free_values[:]
+
+        if cauchy[end] < 1e-5
+            break
+        end
+
+        global γ = update_γ(γ, k+1)
+
+        if γ > 1e11
+            break
+        end
+    end
+    return u, newton_its_u
+end
+
+J = Gridap.Algebra.jacobian( FEOperator(au0, jacu0, Uu, Vu), interpolate_everywhere(x->0.0, Uu))
+
+global uh = FEFunction(Vu, zeros(Vu.nfree))
+global uh_ = uh.free_values[:]
+global Th = FEFunction(VT, ones(VT.nfree))
+
+nls_u = NLSolver(show_trace=false, method=:newton, linesearch=LineSearches.BackTracking(c_1=-1e8), ftol=1e-5, xtol=10*eps())
+solver_u = FESolver(nls_u)
+
+nls_T = NLSolver(show_trace=false, method=:newton, linesearch=LineSearches.BackTracking(c_1=-1e8), ftol=1e-10, xtol=10*eps())
+solver_T = FESolver(nls_T)
+
+newton_its_T, newton_its_u = 0, 0
+cauchy = []
+
+tic = @elapsed for j in 1:10_000
+    print("Considering Fixed Point Iteration: $j. \n")
+    print("     Solving for T.\n")
+    global Th, its = solve!(Th,solver_T,opT)
+    newton_its_T += its.result.iterations
+
+    print("     Solving for u.\n")
+    global γ = 1e0
+    global uh, its_u = path_following_solve(opu, solver_u, uh, J, γ)
+    newton_its_u += its_u
```

Comment:
I'm getting:
```bash
└ @ ~/shared/thermoforming_fixed_point.jl:131
┌ Warning: Assignment to `newton_its_u` in soft scope is ambiguous because a global variable by the same name exists: `newton_its_u` will be treated as a new local. Disambiguate by using `local newton_its_u` to suppress this warning or `global newton_its_u` to assign to the existing global variable.
└ @ ~/shared/thermoforming_fixed_point.jl:136
Considering Fixed Point Iteration: 1. 
     Solving for T.
ERROR: LoadError: UndefVarError: `newton_its_T` not defined in local scope
Stacktrace:
 [1] macro expansion
   @ ~/shared/thermoforming_fixed_point.jl:131 [inlined]
 [2] top-level scope
   @ ./timing.jl:421
```